### PR TITLE
constructors === id

### PR DESCRIPTION
**Pull request by FintanH**
Phase 2 of #244 

Changes `constructors` to the identity function. Marking it as deprecated and a the projection operator `.` should be used instead.

### CODE REVIEW

File: standard/semantics.md [hunk 1]
```diff
@@ -2796,25 +2796,26 @@ union literal:
     merge t₀ u₀ ⇥ merge t₁ u₁
 
 
-`constructors` converts a union type literal to a record of constructors for
-each alternative:
+`[DEPRECTAED]` Union construction should be done through the `.` operator.
```
[the remark this answers is not in the document]
Comment (FintanH) [reply standard/semantics.md]: :flushed:

**Comment by f-f** [comment]
@Gabriel439 since this is technically breaking should we have a (minor) release with #249 first (that is, before merging this) so there's a migration path?

**Comment by Gabriel439** [comment]
@f-f: Alright, then let's defer merging this until after the next release to ease the migration process

**Comment by FintanH** [comment]
@f-f This is good to revisit right? I almost forgot about it until I was telling a friend about Dhall and he saw I had a branch :sweat_smile:

**Comment by FintanH** [comment]
@f-f Added the identity test under `normalization` :smile_cat:

**Comment by FintanH** [comment]
Discussed with @f-f and our plan is to prepare for a release on 19th of Nov since this contains breaking changes. Does this line up with you @Gabriel439?

**Comment by f-f** [comment]
I'll clarify that if the upcoming release is not breaking (i.e. minor) we should release first and then merge, otherwise we'll merge this in the upcoming release.
Another possibility is that we tag the release to https://github.com/dhall-lang/dhall-lang/commit/af53b07523c3d130dc3fbe3faad63439529aac35, but this would probably mean there won't be a paired `dhall-haskell` release

**Comment by Gabriel439** [comment]
@f-f: https://github.com/dhall-lang/dhall-lang/pull/263 is a breaking change that has been introduced since the last release

However, I still think we should cut a release before merging this, regardless of other breaking changes, so that people have a smooth migration path for `constructors`-related code

**Comment by Gabriel439** [comment]
@FintanH: Yeah, feel free to cut a release.  I'm prepared to cut a release of the Haskell implementation at any time.  I usually cut releases for both the standard and the Haskell implementation roughly every 30 days, and it's been almost 30 days since the last release.

**Comment by FintanH** [comment]
Ahhh cool, makes sense! We're not in a rush to get it since we figured out the merging of records issue :)

**Pull request merged**